src/com/generic/util/Persistent.java

```diff
@@ -57,5 +59,40 @@ private void parseWarehouseContentsToObjects(JSONObject shipmentObject) {
 		// add the shipment to the warehouse
 		warehouseTracker.addShipment(warehouseID, shipment);
 	}
-
+	
+	public void parseXML(String filepath) throws IOException, ParseException {
+		File inputFile = new File(filepath);
+		WarehouseTracker warehouseTracker = WarehouseTracker.getInstance();
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+		Document doc = dBuilder.parse(inputFile);
+		doc.getDocumentElement().normalize();
+		NodeList nList = doc.getElementsByTagName("Warehouse");
+		for (int i = 0; i < nList.getLength(); i++) {
+			Node nNode = nList.item(i);
+			if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+				Element eWarehouseElement = (Element) nNode;
+				String warehouseID = eWarehouseElement.getAttribute("id");
+				Warehouse warehouse = new Warehouse(warehouseID);
+				warehouseTracker.addWarehouse(warehouse);
+				if (eWarehouseElement.hasChildNodes()) {
+					NodeList nChildList = eWarehouseElement.getChildNodes();
+					for (int j = 0; j < nChildList.getLength(); j++) {
+						Node nNode2 = nChildList.item(j);
+						if (nNode2.getNodeType() == Node.ELEMENT_NODE) {
+							Element eShipmentElement = (Element) nNode2;
+							Shipment shipment = new Shipment.Builder()
```

Comment:
Also here, a weight unit attribute has been added for this 2nd requirement. Please read the requirements for iteration 2 and make code changes accordingly. Everything else looks alright.